server/src/main/java/org/cloudfoundry/identity/uaa/zone/UserConfigValidator.java

```diff
@@ -0,0 +1,23 @@
+package org.cloudfoundry.identity.uaa.zone;
+
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class UserConfigValidator {
+    private static Logger logger = LoggerFactory.getLogger(UserConfigValidator.class);
+
+    // add a private constructor to hide the implicit public one
+    private UserConfigValidator() {
+    }
+
+    public static void validate(UserConfig config) throws InvalidIdentityZoneConfigurationException {
+        Set<String> allowedGroups = (config == null) ? null : config.resultingAllowedGroups();
+        if ((allowedGroups != null) && (allowedGroups.isEmpty())) {
+            String message = "At least one group must be allowed";
+            logger.error(message);
+            throw new InvalidIdentityZoneConfigurationException(message);
+        }
+    }
+}
```

Comment:
we will have more to validate later, so this validator classes are a pattern in other parts of UAA

Reply:
I would prefer to stay with a separate class UserConfig, because we now have a matching test class UserConfigTest. There are no member variables in UserConfig, so I would like to leave it static